packages/chats/src/BubblePlainText.tsx

```diff
@@ -26,6 +28,29 @@ import { MessageStatus, OnLongPressText, OnPressUrl } from './types';
 import type { ChatPlainTextMessage, PlainTextMessage } from './types';
 import { useBubblePosition, useBubbleContainerStyle } from './useBubblePosition';
 import { useBubbleBackgroundColor, useBubbleRoundedCornerStyle } from './useBubbleStyle';
+import { URI_TON_SURF } from './constants';
+
+const renderParsedText = (matchingString: string, matches: string[]) => {
+    if (matches[1]) {
+        const domain = matches[0].replace(matches[1], '').replace(matches[2], '');
+        if (domain === URI_TON_SURF) {
+            const clearUrl = matches[0].replace(matches[1], '');
+            return matchingString.replace(matches[0], '') + decodeURI(clearUrl);
+        }
+    }
+    return matchingString;
+};
+
+const formatURI = (str: string) => {
+    const encodedURIArray = encodeURI(str).match(PATTERNS.url);
```

Comment:
What if the text is not just an URL, but some text + URL, then in this case `encodeURI` should fail AFAIR.